service/create/launch_configuration.go

```diff
@@ -0,0 +1,146 @@
+package create
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/aws-operator/resources"
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/certificatetpr"
+	cloudconfig "github.com/giantswarm/k8scloudconfig"
+	microerror "github.com/giantswarm/microkit/error"
+
+	awsutil "github.com/giantswarm/aws-operator/client/aws"
+	awsresources "github.com/giantswarm/aws-operator/resources/aws"
+)
+
+type launchConfigurationInput struct {
+	name                string
+	clients             awsutil.Clients
+	cluster             awstpr.CustomObject
+	tlsAssets           *certificatetpr.CompactTLSAssets
+	bucket              resources.Resource
+	securityGroup       resources.ResourceWithID
+	subnet              *awsresources.Subnet
+	keypairName         string
+	instanceProfileName string
+	prefix              string
+}
+
+func (s *Service) createLaunchConfiguration(input launchConfigurationInput) (bool, error) {
+	var (
+		extension    cloudconfig.Extension
+		imageID      string
+		instanceType string
+	)
+
+	switch input.prefix {
+	case prefixMaster:
```

Comment:
I've added master support to the launch config but we still need to decide if we'll use an ASG while we have a single master node. The benefit is we could retired the runMaster code in the create service if we do this.